torch/utils/data/dataloader.py

```diff
@@ -252,23 +258,25 @@ def __init__(self, loader):
             self.worker_init_fn = loader.worker_init_fn
             self.index_queues = [multiprocessing.Queue() for _ in range(self.num_workers)]
             self.worker_queue_idx = 0
-            self.worker_result_queue = multiprocessing.SimpleQueue()
+            self.worker_result_queue = multiprocessing.Queue()
             self.batches_outstanding = 0
             self.worker_pids_set = False
             self.shutdown = False
             self.send_idx = 0
             self.rcvd_idx = 0
             self.reorder_dict = {}
+            self.done_event = multiprocessing.Event()
 
             self.workers = [
                 multiprocessing.Process(
                     target=_worker_loop,
                     args=(self.dataset, self.index_queues[i],
-                          self.worker_result_queue, self.collate_fn, base_seed + i,
+                          self.worker_result_queue, self.done_event,
+                          self.collate_fn, base_seed + i,
                           self.worker_init_fn, i))
                 for i in range(self.num_workers)]
 
-            if self.pin_memory or self.timeout > 0:
+            if self.pin_memory:
                 self.data_queue = queue.Queue()
                 if self.pin_memory:
```

Comment:
This conditional is no longer needed now

Reply:
Also, does the thread that we start here have any other function than pinning memory now? Maybe we should call it `_pin_thread` instead of `worker_manager_thread`, since it doesn't really manage anything